components/EmotionList.js

```diff
@@ -1,17 +1,40 @@
 import styled from "styled-components";
 import Link from "next/link";
 import EmotionCard from "./EmotionCard";
+import { useState } from "react";
+import { emotions } from "@/lib/emotions";
 
 export default function EmotionList({ objects }) {
+  const [selectedFilter, setSelectedFilter] = useState("");
+
+  const filteredObjects = selectedFilter
+    ? objects.filter((item) => item.emotion === selectedFilter)
+    : objects;
+
   return (
     <StyledEmotionList>
-      {objects.length === 0 ? (
+      <StyledHeadline>Filter emotion type</StyledHeadline>
+      <StyledDiv>
```

Comment:
I would suggest to specify the naming of the styled component. Since the StyledDiv is a List of filter selections it could be named as `StyledSelectFilterList` e.g.

Reply:
We didn't even needed this StyledDiv. I've changed it to div and deleted the StyledComponent.